portfolio/src/main/webapp/script.js

```diff
@@ -43,4 +43,14 @@ function addRandomQuote() {
     });
 }
 
+/** Adds an HTML element from Servlet */
+function addElement() {
+  fetch('/data')
+    .then(response => response.text())
+    .then(greeting => {
+      document.getElementById('greeting-container').innerHTML = greeting;
```

Comment:
HTML should almost never be rendered directly from the backend. It exposes the application to attack similar to how you should almost never use JavaScript's `eval()` method. If the request is intercepted, somehow, and the response is replaced with malicious code then it could expose the user.

This isn't a big deal here but it is something to keep in mind.

No action necessary.

Reply:
Thanks for suggestions Cory . 
I will not merge this pull request as this was just one part of the walkthrough. 
json branch is overwriting all these changes.